ThreadSocket/server.c

```diff
@@ -0,0 +1,69 @@
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+char client_message[2000];
+char buffer[1024] = "Message from server";
+pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+void *SocketThread(void *arg) 
+{
+  int newSocket = *((int *)arg);
+  recv(newSocket, client_message, 2000, 0);
+  printf("%s\n", client_message);
+  pthread_mutex_lock(&lock);
+  sleep(1);
+  send(newSocket, buffer, strlen(buffer), 0);
+  printf("Exit socketThread \n");
+  close(newSocket);
+  pthread_exit(NULL);
+}
+
+int main() 
+{
+
+  int serverSocket, newSocket;
+  struct sockaddr_in serverAddr;
+  struct sockaddr_storage serverStorage;
+  socklen_t addr_size;
+
+  serverSocket = socket(PF_INET, SOCK_STREAM, 0);
+
+  serverAddr.sin_family = AF_INET;
+
+  serverAddr.sin_port = htons(7799);
+
+  serverAddr.sin_addr.s_addr = inet_addr("127.0.0.1");
+
+  memset(serverAddr.sin_zero, '\0', sizeof serverAddr.sin_zero);
+
+  bind(serverSocket, (struct sockaddr *)&serverAddr, sizeof(serverAddr));
+
+  listen(serverSocket, 50);
+
+  pthread_t tid[3];
+  int i = 0;
+  while (1) 
+  {
```

Comment:
Let's improve it by adding:
1. When each client type "EXIT" when chatting with server, the connection between server and client will be closed. (end the client process)
2. When all client exit, and no client connect in next 10s, let's close the server, end the server process.